camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/creation/BrooklynEntityDecorationResolver.java

```diff
@@ -230,4 +237,51 @@ protected void addDecorationFromJsonMap(Map<?, ?> decorationJson, List<SpecParam
             throw new IllegalStateException("Not called");
         }
     }
+
+    public static class TagsResolver extends BrooklynEntityDecorationResolver<Iterable<Object>> {
+        protected TagsResolver(BrooklynYamlTypeInstantiator.Factory instantiator) {
+            super(instantiator);
+        }
+
+        @Override
+        public void decorate(EntitySpec<?> entitySpec, ConfigBag attrs, Set<String> encounteredRegisteredTypeIds) {
+            if (getDecorationAttributeJsonValue(attrs) != null) {
```

Comment:
Minor personal preference: I'd store this as a variable rather than evaluating it twice.